### PR TITLE
Feat/nft commands

### PR DESCRIPTION
**Pull request by Kenny-Gin1**
image

### CODE REVIEW

File: apps/web/src/components/commands/nft/Config.tsx [hunk 1]
```diff
@@ -19,17 +19,18 @@ export const NftConfig = (props: CommandExtensionProps<NftWidgetProps>) => {
...
 
     props.editor.view.dom.focus();
   }
 
-  const { property, address, chain } = props.node.attrs;
+  const { property, address, chain, tokenId } = props.node.attrs;
 
   const isConfigured =
-    property !== undefined && address !== undefined && chain !== undefined;
+    property !== undefined && address !== undefined && chain !== undefined || tokenId !== undefined
```
Comment (philipandersson) [inline apps/web/src/components/commands/nft/Config.tsx:33]: Hmm, seems a little hacky. Would be nice in the future to separate the NFT commands configuration. A strategy pattern could solve it.

File: apps/web/src/components/commands/nft/Nft.tsx [hunk 2]
```diff
@@ -27,20 +33,96 @@ const NftFloorWidget = ({
...
+      url,
+      {
+        headers: {
+          Authorization: process.env.NEXT_PUBLIC_NFT_PORT_API_KEY as string,
+        },
+      }
+    );
+    const json = await response.json();
+    return {
+      image: json.nft.cached_file_url as string,
```
Comment (philipandersson) [inline apps/web/src/components/commands/nft/Nft.tsx:122]: This is fragile if the NF doesn't exist or can't be find.

File: apps/web/src/components/commands/nft/Nft.tsx [hunk 3]
```diff
@@ -27,20 +33,96 @@ const NftFloorWidget = ({
...
+    const response = await fetch(
+      `https://api.nftport.xyz/v0/transactions/stats/${address}?chain=${NFT_PORT_CHAINS[chain]}`,
+      {
+        headers: {
+          Authorization: process.env.NEXT_PUBLIC_NFT_PORT_API_KEY as string,
+        },
+      }
+    );
+    const json = await response.json();
+
```
Comment (philipandersson) [inline apps/web/src/components/commands/nft/Nft.tsx:70]: `response.ok` throw error?

File: apps/web/src/components/commands/nft/Nft.tsx [hunk 4]
```diff
@@ -27,20 +33,96 @@ const NftFloorWidget = ({
...
+    const url = `https://api.nftport.xyz/v0/nfts/${address}/${tokenId}?chain=${NFT_PORT_CHAINS[chain]}`
+    const response = await fetch(
+      url,
+      {
+        headers: {
+          Authorization: process.env.NEXT_PUBLIC_NFT_PORT_API_KEY as string,
+        },
+      }
+    );
+    const json = await response.json();
```
Comment (philipandersson) [inline apps/web/src/components/commands/nft/Nft.tsx:120]: `response.ok` throw error?

**Review by philipandersson — APPROVED** [review]
Finally NFTs 🖼️, good job 👏

File: apps/web/src/components/commands/nft/Nft.tsx [hunk 5]
```diff
@@ -1,18 +1,24 @@
...
+  tokenId?: number;
 };
 
 export const NftWidget = ({ property, ...props }: NftWidgetProps) => {
   switch (property) {
     case "floor":
       return <NftFloorWidget {...props} />;
     case "holders":
-      return null;
+      // return <NftHoldersWidget {...props} />;
```
Comment (ericzakariasson) [inline apps/web/src/components/commands/nft/Nft.tsx:17]: Will this case fall through to the next one? 🔍

File: apps/web/src/components/commands/nft/Nft.tsx [hunk 6]
```diff
@@ -27,20 +33,96 @@ const NftFloorWidget = ({
...
     const response = await fetch(
-      `https://api.nftport.xyz/v0/transactions/stats/${address}?chain=${NFT_PORT_CHAINS[chain]}`,
+      url,
       {
         headers: {
           Authorization: process.env.NEXT_PUBLIC_NFT_PORT_API_KEY as string,
         },
       }
     );
+      console.log({ url })
```
Comment (ericzakariasson) [inline apps/web/src/components/commands/nft/Nft.tsx:45]: 🧹

**Review by ericzakariasson — APPROVED** [review]
Nice! 👏 What should we do with the holder command?

**Pull request merged**